core/config.go

```diff
@@ -159,6 +166,46 @@ func (d *Config) BoltOptions() *bolt.Options {
 	return d.boltOpts
 }
 
+// WithDBStorageEngine allows setting the specific storage type
+func WithDBStorageEngine(engine chain.StorageType) ConfigOption {
+	return func(d *Config) {
+		d.dbStorageEngine = engine
+	}
+}
+
+// WithPgDSN applies PosgresSQL specific options to the PG store.
+// It will also create a new database connection.
+func WithPgDSN(dsn string) ConfigOption {
+	return func(d *Config) {
+		d.pgDSN = dsn
+
+		if d.dbStorageEngine != chain.PostgresSQL {
+			// TODO (dlsniper): Would be nice to have a log here. It needs to be injected somehow.
+			return
+		}
+
+		pgConf, err := database.ConfigFromDSN(dsn)
+		if err != nil {
+			panic(err)
+		}
+
+		//nolint:gomnd // We want a reasonable timeout to connect to the database. If it's not done in 5 seconds, then there are bigger problems.
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
Technically, a user can specify this via the dsn. However, it's better to have a failsafe.
I think 5 seconds should be enough for establishing a DB connection. Thoughts?